File: backend/backend-node/scripts/check-env-config.js
/**
 * Script kiểm tra cấu hình environment variables
 * Chạy: node scripts/check-env-config.js
 */

require('dotenv').config({ path: require('path').join(__dirname, '..', '.env') });

const requiredVars = {
    // Server
    PORT: process.env.PORT || '1444',
    NODE_ENV: process.env.NODE_ENV || 'development',
    
    // Database
    MONGO_URI: process.env.MONGO_URI || process.env.MONGODB_URI,
    
    // JWT
    JWT_SECRET: process.env.JWT_SECRET,
    REFRESH_SECRET: process.env.REFRESH_SECRET,
    
    // Security
    PWD_PEPPER: process.env.PWD_PEPPER,
    SESSION_SECRET: process.env.SESSION_SECRET,
    
    // Audio Generation (TTS)
    PYTHON_API_URL: process.env.PYTHON_API_URL || 'http://localhost:8000',
    
    // Conversation Config
    CONVERSATION_MIN_LINES: process.env.CONVERSATION_MIN_LINES || '2',
    CONVERSATION_MAX_LINES: process.env.CONVERSATION_MAX_LINES || '10',
    CONVERSATION_MIN_PARTICIPANTS: process.env.CONVERSATION_MIN_PARTICIPANTS || '2',
    CONVERSATION_MAX_PARTICIPANTS: process.env.CONVERSATION_MAX_PARTICIPANTS || '5'
};

const optionalVars = {
    FRONTEND_URL: process.env.FRONTEND_URL,
    CORS_ALLOWLIST: process.env.CORS_ALLOWLIST,
    SMTP_HOST: process.env.SMTP_HOST,
    SMTP_USER: process.env.SMTP_USER,
    GOOGLE_CLIENT_ID: process.env.GOOGLE_CLIENT_ID
};

console.log('\n🔍 Kiểm tra Environment Variables\n');
console.log('='.repeat(60));

let hasErrors = false;
let hasWarnings = false;

// Check required variables
console.log('\n📋 Required Variables:');
Object.entries(requiredVars).forEach(([key, value]) => {
    if (!value || value === '') {
        console.log(`  ❌ ${key}: MISSING`);
        hasErrors = true;
    } else if (key.includes('SECRET') || key.includes('PEPPER')) {
        // Mask sensitive values
        const masked = value.length > 10 ? value.substring(0, 4) + '...' + value.substring(value.length - 4) : '***';
        console.log(`  ✅ ${key}: ${masked}`);
    } else {
        console.log(`  ✅ ${key}: ${value}`);
    }
});

// Check optional variables
console.log('\n📋 Optional Variables:');
Object.entries(optionalVars).forEach(([key, value]) => {
    if (!value || value === '') {
        console.log(`  ⚠️  ${key}: NOT SET (optional)`);
        hasWarnings = true;
    } else {
        console.log(`  ✅ ${key}: ${value}`);
    }
});

// Check Python API connection
console.log('\n🔗 Python API Connection:');
const pythonApiUrl = process.env.PYTHON_API_URL || 'http://localhost:8000';
console.log(`  URL: ${pythonApiUrl}`);

// Note about Python backend env vars
console.log('\n📝 Lưu ý về Python Backend:');
console.log('  Các API keys sau cần được cấu hình trong backend-python/.env:');
console.log('  - OPENAI_API_KEY (required cho TTS)');
console.log('  - DEEPGRAM_API_KEY (optional, cho Deepgram TTS)');
console.log('  - CONVERSATION_AUDIO_PATH (optional, default: ./conversation_audio)');

console.log('\n' + '='.repeat(60));

if (hasErrors) {
    console.log('\n❌ Có lỗi: Một số biến bắt buộc chưa được cấu hình!');
    console.log('💡 Vui lòng kiểm tra file .env trong thư mục backend-node/');
    process.exit(1);
} else if (hasWarnings) {
    console.log('\n⚠️  Cảnh báo: Một số biến tùy chọn chưa được cấu hình');
    console.log('💡 Hệ thống vẫn có thể hoạt động nhưng một số tính năng có thể bị hạn chế');
    process.exit(0);
} else {
    console.log('\n✅ Tất cả các biến môi trường đã được cấu hình đúng!');
    process.exit(0);
}

File: backend/backend-python/scripts/check-env-config.py
#!/usr/bin/env python3
"""
Script kiểm tra cấu hình environment variables cho Python backend
Chạy: python scripts/check-env-config.py
"""

import os
import sys
from pathlib import Path
from dotenv import load_dotenv

# Load .env từ thư mục hiện tại hoặc parent
env_paths = [
    Path(__file__).parent.parent / '.env',  # backend-python/.env
    Path(__file__).parent / '.env',         # app/.env
    Path.cwd() / '.env',                    # Current working directory
]

env_loaded = False
for env_path in env_paths:
    if env_path.exists():
        load_dotenv(env_path)
        print(f"✅ Loaded .env from: {env_path}")
        env_loaded = True
        break

if not env_loaded:
    print("⚠️  No .env file found!")
    print("💡 Please create .env file from env.example:")
    print("   cp env.example .env")

required_vars = {
    'OPENAI_API_KEY': os.getenv('OPENAI_API_KEY'),
    'HOST': os.getenv('HOST', '0.0.0.0'),
    'PORT': os.getenv('PORT', '8000'),
}

optional_vars = {
    'DEEPGRAM_API_KEY': os.getenv('DEEPGRAM_API_KEY'),
    'CONVERSATION_AUDIO_PATH': os.getenv('CONVERSATION_AUDIO_PATH'),
    'AUDIO_STORAGE_PATH': os.getenv('AUDIO_STORAGE_PATH'),
    'ALLOWED_ORIGINS': os.getenv('ALLOWED_ORIGINS'),
    'DEBUG': os.getenv('DEBUG', 'false'),
}

print('\n' + '=' * 60)
print('🔍 Kiểm tra Environment Variables cho Python Backend')
print('=' * 60)

has_errors = False
has_warnings = False

# Check required variables
print('\n📋 Required Variables:')
for key, value in required_vars.items():
    if not value or value == '':
        print(f'  ❌ {key}: MISSING')
        has_errors = True
    elif key == 'OPENAI_API_KEY':
        # Mask API key
        masked = value[:7] + '...' + value[-4:] if len(value) > 11 else '***'
        print(f'  ✅ {key}: {masked}')
    else:
        print(f'  ✅ {key}: {value}')

# Check optional variables
print('\n📋 Optional Variables:')
for key, value in optional_vars.items():
    if not value or value == '':
        print(f'  ⚠️  {key}: NOT SET (optional)')
        has_warnings = True
    else:
        if key == 'DEEPGRAM_API_KEY':
            masked = value[:7] + '...' + value[-4:] if len(value) > 11 else '***'
            print(f'  ✅ {key}: {masked}')
        else:
            print(f'  ✅ {key}: {value}')

# Check OpenAI API key format
openai_key = os.getenv('OPENAI_API_KEY', '')
if openai_key:
    if not openai_key.startswith('sk-'):
        print('\n⚠️  WARNING: OPENAI_API_KEY không đúng format (nên bắt đầu với "sk-")')
        has_warnings = True

# Check Deepgram API key format
deepgram_key = os.getenv('DEEPGRAM_API_KEY', '')
if deepgram_key:
    if len(deepgram_key) < 20:
        print('\n⚠️  WARNING: DEEPGRAM_API_KEY có vẻ không đúng format')
        has_warnings = True

print('\n' + '=' * 60)

if has_errors:
    print('\n❌ Có lỗi: Một số biến bắt buộc chưa được cấu hình!')
    print('💡 Vui lòng kiểm tra file .env trong thư mục backend-python/')
    sys.exit(1)
elif has_warnings:
    print('\n⚠️  Cảnh báo: Một số biến tùy chọn chưa được cấu hình')
    print('💡 Hệ thống vẫn có thể hoạt động nhưng một số tính năng có thể bị hạn chế')
    sys.exit(0)
else:
    print('\n✅ Tất cả các biến môi trường đã được cấu hình đúng!')
    sys.exit(0)
